### ISSUE TITLE
Better way to handle free disk space calculation on macOS + APFS

### ISSUE THREAD

**Issue by luckman212** [syncthing/syncthing#9737]
### Issue

Currently (as of v1.27.12) syncthing is unable to correctly identify the amount of free space on APFS-formatted volumes. This causes unnecessary errors and sync failures if the `minDiskFree` and/or `minHomeDiskFree` values are not set to `0`. For example, this happened to me this morning:

![image](https://github.com/user-attachments/assets/cc04ccd3-00b0-48e6-89df-ef82fd390abb)

### Idea

Apple has a [public API][1] that can be used to correctly report free space. Here's an example POC program written in Swift: https://github.com/luckman212/apfs-free

I'm not sure if or how this could be integrated into the Go code. We could "shell out", but that might be too complex or expensive. Just opening this for dicussion... if a fix isn't possible, I believe this should at least be added to the [FAQ][2].

### Related

- https://github.com/syncthing/syncthing/issues/5976
- Forum post: [“insufficient space on disk for database”](https://forum.syncthing.net/t/21717)

[1]: https://developer.apple.com/documentation/foundation/urlresourcekey/checking_volume_storage_capacity
[2]: https://docs.syncthing.net/users/faq.html

### Problem or use case

-

### Alternatives or workarounds

-

**Comment by calmh** [comment]
I guess "correct" is in the eye of the beholder here. Syncthing reacts to the actual free space on disk. Finder reports free space plus used space that could be freed if it felt the need to do so, which it apparently hasn't, yet.

**Comment by luckman212** [comment]
Yeah I hate all the obfuscation too. I shouldn't have said "correctly", maybe "confidently" describes it better? I don't know, APFS is just a mystery box. So, maybe just a FAQ entry to suggest that on macOS, this setting should "probably" be set to 0.
